packages/core/src/App/Containers/Layout/header/dtrader-header.jsx

```diff
@@ -0,0 +1,347 @@
+                    {menu_items && is_logged_in && replaceCashierMenuOnclick()}
+                    <MenuLinks is_logged_in={is_logged_in} items={menu_items} />
+                </div>
+
+                <div
+                    className={classNames('header__menu-right', {
+                        'header__menu-right--hidden': isMobile() && is_logging_in,
+                    })}
+                >
+                    <DesktopWrapper>
+                        <RedirectToOldInterface />
+                        <div className='header__menu--dtrader--separator--account'>
+                            <Divider />
+                        </div>
+                    </DesktopWrapper>
+                    {is_logging_in && (
+                        <div
+                            id='dt_core_header_acc-info-preloader'
+                            className={classNames('acc-info__preloader__dtrader', {
+                                'acc-info__preloader__dtrader--no-currency': !currency,
+                                'acc-info__preloader__dtrader--is-crypto': getDecimalPlaces(currency) > 2,
+                            })}
+                        >
+                            <AccountsInfoLoader is_logged_in={is_logged_in} is_mobile={isMobile()} speed={3} />
+                        </div>
+                    )}
+                    <div id={'dt_core_header_acc-info-container'} className='acc-info__container'>
+                        <AccountActions
+                            acc_switcher_disabled_message={acc_switcher_disabled_message}
+                            account_type={account_type}
+                            balance={balance}
+                            currency={currency}
+                            country_standpoint={country_standpoint}
+                            disableApp={disableApp}
+                            enableApp={enableApp}
+                            is_acc_switcher_on={is_acc_switcher_on}
+                            is_acc_switcher_disabled={is_acc_switcher_disabled}
+                            is_eu={is_eu}
+                            is_notifications_visible={is_notifications_visible}
+                            is_logged_in={is_logged_in}
+                            is_virtual={is_virtual}
+                            onClickDeposit={onClickDeposit}
+                            notifications_count={notifications_count}
+                            toggleAccountsDialog={toggleAccountsDialog}
+                            toggleNotifications={toggleNotifications}
+                            openRealAccountSignup={openRealAccountSignup}
+                        />
+                    </div>
+                </div>
+            </div>
+            <RealAccountSignup />
+            <SetAccountCurrencyModal />
+            <NewVersionNotification onUpdate={addUpdateNotification} />
+            <TempAppSettings />
+        </header>
+    );
+};
+
+DTraderHeader.propTypes = {
```

Comment:
Since using typescript for this component requires the extra time of its own, I think it would be better if you leave it as it is.